apis/core/v1alpha2/feature_types.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FeatureSpec defines the desired state of Feature
+type FeatureSpec struct {
+	// Description of the feature.
+	Description string `json:"description"`
+	// Stability indicates stability level of the feature.
+	// Stability levels are Work In Progress, Experimental, Technical Preview, Stable and Deprecated.
+	// +kubebuilder:validation:Enum=Work In Progress;Experimental;Technical Preview;Stable;Deprecated
+	// - Work In Progress: Feature is not ready, it is still being developed
+	// - Experimental: Feature is not ready, but customers can try it out in pre-production clusters. It's unsupported and voids warranty.
+	// - Technical Preview: Almost ready, but customers are encouraged to try it out in pre-production clusters. It's unsupported, but doesn't void warranty
+	// - Stable: Feature is ready and fully supported
+	// - Deprecated: Feature is destined for removal, usage is discouraged
```

Comment:
How about:

> Feature is destined for removal, usage is discouraged. Deactivate this feature prior to upgrading to a release which has removed it to validate that you are not still using it and to prevent users from introducing new usage of it.

(Disclaimer: I feel like this still isn't as clear as it could be.)

Reply:
i think this is clear